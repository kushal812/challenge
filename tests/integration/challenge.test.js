const request = require('supertest');
const httpStatus = require('http-status');
const app = require('../../src/app');
const setupTestDB = require('../utils/setupTestDB');

setupTestDB();

describe('Challenge routes', () => {
  describe('POST /v1/challenge', () => {
    let query;
    beforeEach(() => {
      query = {
        startDate: '2016-01-26',
        endDate: '2018-02-02',
        minCount: 1,
        maxCount: 10,
      };
    });
    test('should return 200 and apply the default query options', async () => {
      const res = await request(app).post('/v1/challenge').send(query).expect(httpStatus.OK);
      expect(res.body).toMatchSnapshot();
    });

    test('should return 400 error if startDate is invalid', async () => {
      query.startDate = 'invalid';
      await request(app).post('/v1/challenge').send(query).expect(httpStatus.BAD_REQUEST);
    });

    test('should return 400 error if endDate is invalid', async () => {
      query.endDate = 'invalid';
      await request(app).post('/v1/challenge').send(query).expect(httpStatus.BAD_REQUEST);
    });

    test('should return 400 error if minCount is invalid', async () => {
      query.minCount = 'invalid';
      await request(app).post('/v1/challenge').send(query).expect(httpStatus.BAD_REQUEST);
    });

    test('should return 400 error if maxCount is invalid', async () => {
      query.maxCount = 'invalid';
      await request(app).post('/v1/challenge').send(query).expect(httpStatus.BAD_REQUEST);
    });
  });
});
