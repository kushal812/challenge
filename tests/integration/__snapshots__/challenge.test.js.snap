// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge routes POST /v1/challenge should return 200 and apply the default query options 1`] = `
Object {
  "code": 0,
  "msg": "Success",
  "records": Array [
    Object {
      "createdAt": "2016-10-06T09:03:06.664Z",
      "key": "XVSJAafS",
      "totalCount": 7,
    },
    Object {
      "createdAt": "2016-06-13T14:21:11.684Z",
      "key": "HFrLrkmu",
      "totalCount": 4,
    },
    Object {
      "createdAt": "2016-05-04T05:15:21.742Z",
      "key": "nksTNdZi",
      "totalCount": 7,
    },
  ],
}
`;
